pkg/async/schedule/factory.go

```diff
@@ -42,21 +44,36 @@ func (w *workflowScheduler) GetEventScheduler() interfaces.EventScheduler {
 }
 
 func (w *workflowScheduler) GetWorkflowExecutor(
+	db repositories.RepositoryInterface,
```

Comment:
this seems like a very broad interface that we are passing here? Can we reduce it? If you want you can create a local interface too